### PR TITLE
ci: update release workflow

### LINKED ISSUE #282: Need for urgent changes in GitHub Actions automation
This issue defines a list of tasks that need to be performed in this repo to make sure it's ci/cd automation works long term without any issues.

It is up to maintainers to decide if it must be addressed in one or multiple PRs.

Below are 3 different sections describing 3 different important ci/cd changes.

**IMPORTANT-START** 
For GitHub workflows that contain `This workflow is centrally managed in https://github.com/asyncapi/.github/` you do not have to perform any work. These workflows were already updated through the update in `.github`. The only exception is the workflows related to nodejs release. More details in **Upgrade Release pipeline - in case of nodejs projects** section
**IMPORTANT-END** 

### Deprecation of way data is shared between steps

Every single GitHub Action workflow that has `echo "::set-output name={name}::{value}"` need to be updated to follow `echo "{name}={value}" >> $GITHUB_OUTPUT`

We do not yet know when `set-output` will stop working. Previous disable date was 31.05 but now then say community needs more time.

[For more details read official article from GitHub](https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/)

### Deprecation of node12

> 2nd bullet point is still relevant for you even if your projects in not nodejs project

- Every single workflow that uses [setup-node](https://github.com/actions/setup-node) action needs an update to follow `v3` version of this action, and make sure minimum node 14 is used
- Now this part is more complex. Problem with `node12` is that node-based GitHub Actions were using it in majority as a runtime environment. Look for example at [this action.yaml file for setup-node action v2](https://github.com/actions/setup-node/blob/v2/action.yml#L38). So the job that you have to do is go through all the workflows, and verify every single action that you use, make sure you are using the latest version that is not based on `node12`. I already did review a lot of actions as part of [this PR](https://github.com/asyncapi/.github/pull/226) so maybe you will find some actions there and can copy from me. For example `action/checkout` needs to be updated to v3.

Node12 end of support in action is probably September 27th.

[For more details read official article from GitHub](https://github.blog/changelog/2022-09-22-github-actions-all-actions-will-begin-running-on-node16-instead-of-node12/)

### Upgrade Release pipeline - in case of nodejs projects

> ignore this section if your project is not nodejs project

You have 2 options. You can:

**A.** choose to switch to new release pipeline using instruction from https://github.com/asyncapi/.github/issues/205

**B.** stay with old release pipeline, and manually update GitHub workflows and actions used in it, you can inspire a lot from this PR https://github.com/asyncapi/.github/pull/226

I definitely recommend going with **A**

Workflows related to release:
- .github/workflows/if-nodejs-release.yml
- .github/workflows/if-nodejs-version-bump.yml
- .github/workflows/bump.yml

### PR DESCRIPTION
**Pull request by princerajpoot20**
**Description**

Update release workflow.

**Related issue(s)**

Fixes #282 

cc @derberg

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -36,8 +36,7 @@
     "lint:fix": "eslint --max-warnings 0 --config .eslintrc . --fix",
     "generate:assets": "npm run generate:readme:toc",
     "generate:readme:toc": "markdown-toc -i README.md",
-    "generate:docker": "docker build --tag asyncapi/server-api .",
-    "release": "npm run generate:docker && semantic-release"
+    "generate:docker": "docker build --tag asyncapi/server-api ."
```
Comment (derberg) [inline package.json:39]: now I noticed that release workflows here are mixed with docker release. There are some additional changes that you need to do, sorry about that

```suggestion
    "docker:build": "docker build --tag asyncapi/server-api ."
```

you also need to add a new file, called `release-docker.yml` like below"
```yml
name: Release Docker Image
on: 
  release:
    types: 
      - published
jobs:

  publish-docker:
    name: Generating Docker
    runs-on: ubuntu-latest
    steps:
      - name: Checkout repository
        uses: actions/checkout@v3
      - name: Get version without v character
        id: version
        run: |
          VERSION=${{github.event.release.tag_name}}
          VERSION_WITHOUT_V=${VERSION:1}
          echo "value=${VERSION_WITHOUT_V}" >> $GITHUB_OUTPUT

      - name: Release to Docker
        run: | 
          echo ${{secrets.DOCKER_PASSWORD}} | docker login -u ${{secrets.DOCKER_USERNAME}} --password-stdin
          npm run docker:build
          docker tag asyncapi/server-api:latest asyncapi/server-api:${{ steps.version.outputs.value }}
          docker push asyncapi/server-api:${{ steps.version.outputs.value }}
          docker push asyncapi/server-api:latest

      - uses: meeDamian/sync-readme@82715041300710d9be7c726c9d6c683b70451087 #version 1.0.6 https://github.com/meeDamian/sync-readme/releases/tag/v1.0.6
        with:
          user: ${{secrets.DOCKER_USERNAME}}
          pass: ${{ secrets.DOCKER_PASSWORD }}
          slug: asyncapi/server-api
          description: CLI to work with your AsyncAPI files
```

**Comment by princerajpoot20** [comment]
@derberg made the changes 🙂

File: package.json [hunk 2]
```diff
@@ -79,9 +78,6 @@
   },
   "devDependencies": {
     "@semantic-release-plus/docker": "^3.1.0",
```
Comment (derberg) [inline package.json:80]: you can remove this too, and refresh package-lock

Comment (princerajpoot20) [reply package.json]: @derberg Done 👍 . Sorry for the delay

**Review by derberg — APPROVED** [review]
well done 👏🏼 

@BOLT04 you can approve and merge

File: package.json [hunk 3]
```diff
@@ -36,8 +36,7 @@
     "lint:fix": "eslint --max-warnings 0 --config .eslintrc . --fix",
     "generate:assets": "npm run generate:readme:toc",
     "generate:readme:toc": "markdown-toc -i README.md",
-    "generate:docker": "docker build --tag asyncapi/server-api .",
-    "release": "npm run generate:docker && semantic-release"
+    "docker:build": "docker build --tag asyncapi/server-api ."
```
Comment (derberg) [inline package.json:39]:
```suggestion
    "docker:build": "docker build --tag asyncapi/server-api ."
    "bump:version": "npm --no-git-tag-version --allow-same-version version $VERSION",
```

please commit it as this is one more missing script that is required

Comment (princerajpoot20) [reply package.json]: @derberg Done

**Review by derberg — APPROVED** [review]
@BOLT04 @smoya ready to merge from my side

**Review by smoya — APPROVED** [review]

**Pull request merged**